src/core/tokenize/match.ts

```diff
@@ -46,11 +42,21 @@ export function _matchGrammar (
 
 			const patternObj = toGrammarToken(patterns[j]);
 			let { pattern, lookbehind = false, greedy = false, alias, inside } = patternObj;
-			const insideGrammar = resolve(prism.languageRegistry, inside);
+			const insideGrammar = resolve.call(prism, inside);
+
+			let flagsToAdd = '';
 
 			if (greedy && !pattern.global) {
 				// Without the global flag, lastIndex won't work
-				patternObj.pattern = pattern = RegExp(pattern.source, pattern.flags + 'g');
+				flagsToAdd += 'g';
+			}
+			if (pattern.source.indexOf('(?<') && pattern.hasIndices === false) {
```

Comment:
Looking at the expression, it seems that `includes` is intended to be used instead of `indexOf`, no?

Reply:
Yes